kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -557,10 +562,25 @@ private void enableInterface(String deviceId, NetworkProperties properties, Devi
         }
 
         try {
+            this.modemManager.failedResetHandlersDisable(deviceId);
             this.networkManager.activateConnection(connection.get(), device);
             dsLock.waitForSignal();
         } catch (DBusExecutionException e) {
             logger.warn("Couldn't complete activation of {} interface, caused by:", deviceId, e);
+            // If a modem connection fails at the first try, it stays in the failed state, thus not triggering the usual
+            // modem reset procedure. So, start a reset timer here.
+            if (isModemFailed(device, deviceType)) {
+                int delayMinutes = properties.get(Integer.class, "net.interface.%s.config.resetTimeout", deviceId);
+
+                if (delayMinutes != 0) {
+                    Optional<String> mmDbusPath = this.networkManager.getModemManagerDbusPath(device.getObjectPath());
+
+                    logger.info("Modem {} in failed state or unavailable. Scheduling modem reset in {} minutes ...",
+                            device.getObjectPath(), delayMinutes);
+
+                    this.modemManager.failedResetHandlerEnable(deviceId, mmDbusPath, delayMinutes);
+                }
+            }
```

Comment:
Here we have a lot of stuff that could throw an exception and is specific for modem devices.

Would it be better to have a `activationFailed` flag on which we can structure our code outside of this `catch` block? Bonus points if we can manage this in the https://github.com/eclipse-kura/kura/blob/e6c27549de30c95e2c3926e0cecbf7c9253c52ba/kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java#L594 block.

```suggestion
            activationFailed = true;
```

Reply:
I like it. By the way, it simplifies the management of the missing modem dbus path and makes the code more clear.